compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1184,15 +1184,39 @@ class TreeUnpickler(reader: TastyReader,
             case SELECTin =>
               var sname = readName()
               val qual = readTerm()
-              val owner = readType()
-              def select(name: Name, denot: Denotation) =
-                val prefix = ctx.typeAssigner.maybeSkolemizePrefix(qual.tpe.widenIfUnstable, name)
-                makeSelect(qual, name, denot.asSeenFrom(prefix))
-              sname match
-                case SignedName(name, sig, target) =>
-                  select(name, owner.decl(name).atSignature(sig, target))
-                case name =>
-                  select(name, owner.decl(name))
+              val ownerTpe = readType()
+              val owner = ownerTpe.typeSymbol
+              val SignedName(name, sig, target) = sname: @unchecked // only methods with params use SELECTin
+              val qualType = qual.tpe.widenIfUnstable
+              val prefix = ctx.typeAssigner.maybeSkolemizePrefix(qualType, name)
+
+              def overloadNotFound(select: Tree, symPart: String) =
+                typer.ErrorReporting.errorTree(select,
+                  i"""While unpickling, I was looking in `$qual` (of widened type: ${qual.tpe.widen})
+                     |but could not find method `$name` (${sname.info}):
+                     | - $symPart.""".stripMargin)
+
+              val denot =
+                val cls = ownerTpe.classSymbol
+                if !cls.exists then NoDenotation
+                else
+                  val clsDenot = cls.asClass.classDenot
+                  val d0 = clsDenot.findMember(name, cls.thisType, EmptyFlags, EmptyFlags).atSignature(sig, target)
+                  if !d0.symbol.exists || d0.symbol.isAccessibleFrom(prefix) then
+                    d0.asSeenFrom(prefix)
+                  else
+                    clsDenot.findMember(name, cls.thisType, EmptyFlags, excluded=Private)
+                      .atSignature(sig, target)
+                      .asSeenFrom(prefix)
+
+              val select = makeSelect(qual, name, denot)
+              if !(owner.exists && denot.exists) && ctx.settings.YstrictTasty.value then
```

Comment:
I'm not a big fan of adding a special path which is only used in tests, since by definition it means that what we're testing doesn't match reality. Can't the tests be adapted to not rely on this?

Reply:
this is no longer needed